mlflow/tensorflow/__init__.py

```diff
@@ -887,14 +897,113 @@ def _patch_implementation(
                     # & tf.keras callbacks if necessary
                     callbacks = list(kwargs.get("callbacks") or [])
                     kwargs["callbacks"], self.log_dir = _setup_callbacks(
-                        callbacks, log_models, metrics_logger
+                        callbacks, False, metrics_logger
                     )
 
                 early_stop_callback = _get_early_stop_callback(callbacks)
                 _log_early_stop_callback_params(early_stop_callback)
 
                 history = original(inst, *args, **kwargs)
 
+                if log_models:
```

Comment:
```suggestion
TF_SUPPORTED_AUTOLOGGING_TYPES = (np.ndarray, 
                  pd.Series, 
                  tensorflow.data.Dataset, 
                  dict, 
                  tensorflow.keras.utils.Sequence)
def _is_supported_for_autologging(input_data):
    support = isinstance(input_data, TF_SUPPORTED_AUTOLOGGING_TYPES)
    if support:
        if isinstance(input_data, dict):
            support = all({isinstance(value, np.ndarray) for value in input_data.values()})
        if isinstance(input_data, tensorflow.keras.utils.Sequence):
            support = isinstance(input_data[:][0], np.ndarray)
    if not support:
        warnings.warn("Tensorflow keras autologging only supports input types of: "
                      "numpy.ndarray, dict(<key> -> numpy.ndarray), tensorflow.data.Dataset, "
                      "or tensorflow.keras.utils.Sequence")
    return support
```
might want to simplify this logic a bit.

Reply:
Incorporated Corey's suggestion below and removed this method. thanks.